File: lib/core/models/day.dart
enum Day {
  sun,
  mon,
  tue,
  wed,
  thu,
  fri,
  sat;

  String toJson() {
    return toString().split('.').last;
  }

  static Day fromJson(String jsonString) {
    return Day.values
        .firstWhere((v) => v.toString().split('.').last == jsonString);
  }
}

File: ios/CoursesWidget/Models/Day.swift
//
//  DayEnum.swift
//  CoursesWidgetExtension
//
//  Created by Yazeed AlKhalaf on 11/03/2023.
//

import Foundation

enum Day: String, Decodable, CodingKey {
    case sun = "sun"
    case mon = "mon"
    case tue = "tue"
    case wed = "wed"
    case thu = "thu"
    case fri = "fri"
    case sat = "sat"
}
